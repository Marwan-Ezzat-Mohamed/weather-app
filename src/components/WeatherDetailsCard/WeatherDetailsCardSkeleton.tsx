import { WeatherInformationItemSkeleton } from "./WeatherInformationItem/WeatherInformationItemSkeleton";
import { WeatherForecastItemSkeleton } from "./WeatherForecastItem/WeatherForecastItemSkeleton";

const WeatherDetailsCardSkeleton = () => {
  return (
    <div className="flex w-full justify-center rounded-2xl bg-slate-100 p-5 transition duration-200 hover:shadow-2xl dark:bg-slate-800">
      <div className="w-full space-y-5">
        <div className="flex justify-between text-2xl">
          <div className="h-6 w-1/3 animate-pulse rounded-md bg-gray-300 dark:bg-gray-500 "></div>
          <div className="h-6 w-1/3 animate-pulse rounded-md bg-gray-300 dark:bg-gray-500"></div>
        </div>
        <section className="flex justify-around">
          <div className="flex flex-col items-center">
            <div className="flex items-center">
              <div className="h-24 w-24 animate-pulse rounded-full bg-gray-300 dark:bg-gray-500"></div>
              <div className="ml-4 h-32 w-32 animate-pulse rounded-md bg-gray-300 dark:bg-gray-500"></div>
            </div>
          </div>
          <div className="flex flex-col text-2xl">
            <div className="mb-5 flex justify-between gap-5">
              <div className="h-6 w-2/3 animate-pulse rounded-md bg-gray-300 dark:bg-gray-500"></div>
              <div className="h-6 w-1/3 animate-pulse rounded-md bg-gray-300 dark:bg-gray-500"></div>
            </div>
            <div className="flex flex-col gap-2">
              <WeatherInformationItemSkeleton />
              <WeatherInformationItemSkeleton />
              <WeatherInformationItemSkeleton />
              <WeatherInformationItemSkeleton />
            </div>
          </div>
        </section>
        <section className="flex flex-col gap-2">
          <span className="h-7 w-2/3 animate-pulse rounded bg-gray-400 dark:bg-gray-500"></span>

          <div className="flex flex-wrap justify-center gap-5">
            <WeatherForecastItemSkeleton />
            <WeatherForecastItemSkeleton />
            <WeatherForecastItemSkeleton />
            <WeatherForecastItemSkeleton />
            <WeatherForecastItemSkeleton />
            <WeatherForecastItemSkeleton />
          </div>
        </section>
      </div>
    </div>
  );
};

export default WeatherDetailsCardSkeleton;
