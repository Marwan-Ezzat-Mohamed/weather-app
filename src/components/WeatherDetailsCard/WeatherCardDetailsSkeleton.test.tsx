import { render } from "@testing-library/react";
import WeatherDetailsCardSkeleton from "./WeatherDetailsCardSkeleton";

test("WeatherCardSkeleton matches snapshot", () => {
  const { container } = render(<WeatherDetailsCardSkeleton />);
  expect(container).toMatchSnapshot();
});
