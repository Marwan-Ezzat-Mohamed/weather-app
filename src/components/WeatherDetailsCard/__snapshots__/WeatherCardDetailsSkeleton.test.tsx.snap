// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherCardSkeleton matches snapshot 1`] = `
<div>
  <div
    class="flex w-full justify-center rounded-2xl bg-slate-100 p-5 transition duration-200 hover:shadow-2xl dark:bg-slate-800"
  >
    <div
      class="w-full space-y-5"
    >
      <div
        class="flex justify-between text-2xl"
      >
        <div
          class="h-6 w-1/3 animate-pulse rounded-md bg-gray-300 dark:bg-gray-500 "
        />
        <div
          class="h-6 w-1/3 animate-pulse rounded-md bg-gray-300 dark:bg-gray-500"
        />
      </div>
      <div
        class="flex justify-around"
      >
        <div
          class="flex flex-col items-center"
        >
          <div
            class="flex items-center"
          >
            <div
              class="h-24 w-24 animate-pulse rounded-full bg-gray-300 dark:bg-gray-500"
            />
            <div
              class="ml-4 h-32 w-32 animate-pulse rounded-md bg-gray-300 dark:bg-gray-500"
            />
          </div>
        </div>
        <div
          class="flex flex-col text-2xl"
        >
          <div
            class="mb-5 flex justify-between gap-5"
          >
            <div
              class="h-6 w-2/3 animate-pulse rounded-md bg-gray-300 dark:bg-gray-500"
            />
            <div
              class="h-6 w-1/3 animate-pulse rounded-md bg-gray-300 dark:bg-gray-500"
            />
          </div>
          <div
            class="flex flex-col gap-2"
          >
            <div
              class="flex justify-between gap-5 "
            >
              <div
                class="flex gap-2 "
              >
                <div
                  class="h-6 w-6 animate-pulse rounded-full bg-gray-300 dark:bg-gray-500"
                />
                <div
                  class="h-6 w-16 animate-pulse rounded-md bg-gray-300 dark:bg-gray-500"
                />
              </div>
              <div
                class="h-6 w-16 animate-pulse rounded-md bg-gray-300 dark:bg-gray-500"
              />
            </div>
            <div
              class="flex justify-between gap-5 "
            >
              <div
                class="flex gap-2 "
              >
                <div
                  class="h-6 w-6 animate-pulse rounded-full bg-gray-300 dark:bg-gray-500"
                />
                <div
                  class="h-6 w-16 animate-pulse rounded-md bg-gray-300 dark:bg-gray-500"
                />
              </div>
              <div
                class="h-6 w-16 animate-pulse rounded-md bg-gray-300 dark:bg-gray-500"
              />
            </div>
            <div
              class="flex justify-between gap-5 "
            >
              <div
                class="flex gap-2 "
              >
                <div
                  class="h-6 w-6 animate-pulse rounded-full bg-gray-300 dark:bg-gray-500"
                />
                <div
                  class="h-6 w-16 animate-pulse rounded-md bg-gray-300 dark:bg-gray-500"
                />
              </div>
              <div
                class="h-6 w-16 animate-pulse rounded-md bg-gray-300 dark:bg-gray-500"
              />
            </div>
            <div
              class="flex justify-between gap-5 "
            >
              <div
                class="flex gap-2 "
              >
                <div
                  class="h-6 w-6 animate-pulse rounded-full bg-gray-300 dark:bg-gray-500"
                />
                <div
                  class="h-6 w-16 animate-pulse rounded-md bg-gray-300 dark:bg-gray-500"
                />
              </div>
              <div
                class="h-6 w-16 animate-pulse rounded-md bg-gray-300 dark:bg-gray-500"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
